be/src/main/java/yeonba/be/util/JwtUtil.java

```diff
@@ -0,0 +1,73 @@
+package yeonba.be.util;
+
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.SignatureException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Date;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import yeonba.be.user.entity.User;
+
+@Component
+public class JwtUtil {
+
+    private final Duration ACCESS_TOKEN_DURATION = Duration.of(8, ChronoUnit.HOURS);
+    private final Duration REFRESH_TOKEN_DURATION = Duration.of(10, ChronoUnit.DAYS);
+
+    @Value("${JWT_SECRET}")
+    private String jwtSecret;
+
+    public String generateAccessToken(User user, Date issuedAt) {
+
+        Date expiredAt = getExpiredAt(issuedAt, ACCESS_TOKEN_DURATION);
+
+        return generateUserJwt(user, issuedAt, expiredAt);
+    }
+
+    public String generateRefreshToken(User user, Date issuedAt) {
+
+        Date expiredAt = getExpiredAt(issuedAt, REFRESH_TOKEN_DURATION);
+
+        return generateUserJwt(user, issuedAt, expiredAt);
+    }
+
+    private Date getExpiredAt(Date issuedAt, Duration duration) {
+
+        Instant instant = issuedAt.toInstant()
+            .plusMillis(duration.toMillis());
+
+        return Date.from(instant);
+    }
+
+    private String generateUserJwt(
+        User user,
+        Date issuedAt,
+        Date expiredAt) {
+
+        return Jwts.builder()
+            .setSubject(user.getEmail())
+            .setIssuedAt(issuedAt)
+            .setExpiration(expiredAt)
+            .claim("userId", user.getId())
+            .signWith(SignatureAlgorithm.HS256, jwtSecret)
+            .compact();
+    }
+
+    public void validateJwt(String jwt) {
+
+        try {
+            Jwts.parser()
+                .setSigningKey(jwtSecret)
+                .parseClaimsJws(jwt);
+
+        } catch (SignatureException e) {
+            throw new IllegalStateException("유효하지 않은 JWT 시그니처입니다.", e);
+        } catch (ExpiredJwtException e) {
+            throw new IllegalStateException("만료된 JWT입니다.", e);
+        }
```

Comment:
제가 이해를 잘못한 것인지 모르겠지만 서버에서 발생하는 에러는 전부 다 GeneralException으로 처리하자고 말씀하셨는데, 이 부분만 다르게  처리하신 이유는 무엇인가요?

추가로, 제 생각에는 저희 API를 제 3의 서비스가 사용하는 것이 아니기 때문에 구체적인 예외 메시지(시그니처가 잘못되었습니다, 만료된 JWT입니다 등)를 응답해줄 필요가 없을 것같은데 아래와 같이 처리하는 것에 대해서 어떻게 생각하시나요?
```suggestion
        } catch (SignatureException, ExpiredJwtException e) {
            throw new IllegalStateException("유효하지 않은 JWT입니다. 다시 로그인 해주세요.", e);
        }
```

Reply:
유틸 기능은 어떤 기술에 의존함에 따라 변경 가능성이 잦은 기능이라고 생각했습니다. 그런 측면에서 유틸 기능에서 발생할 수 있는 예외를 도메인 연관 예외와 같은 형식으로 처리하는 것이 적절한 지 의문이 들어 일단 저렇게 정의하였습니다.

Reply:
util 패키지에서 발생했다면 UtilException과 같은 형태로 일관성 있게 맞춰도 차이가 없는 것같다는 생각이 드네요.
그리고 말씀하신 내용에서 `의존`이라는 단어를 사용하셨는데 의존함에 따라 변경이 잦아진다면 적절하게 역할을 분리한 것인지에 대해서 조금 더 고민을 해보는 것이 좋지 않을까 생각이 듭니다.

Reply:
`UtilException`으로 일관성 있게 처리하는 것에 동의합니다. 반영하겠습니다.